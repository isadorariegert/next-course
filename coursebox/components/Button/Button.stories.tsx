import { ComponentStoryObj, ComponentMeta } from '@storybook/react';
import { screen, userEvent } from '@storybook/testing-library';
import { expect } from '@storybook/jest';
import { Button } from './Button';

export default {
  title: 'Controls/Button',
  component: Button,
  args: {
    children: 'Button',
  },
} as ComponentMeta<typeof Button>;

// const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const PrimaryButton: ComponentStoryObj<typeof Button> = {
  play: async ({ args }) => {
    await userEvent.click(screen.getByRole('button'));
    await userEvent.click(screen.getByRole('button'));
    // @todo: https://github.com/storybookjs/storybook/issues/16941
    await expect(args.onClick).toHaveBeenCalledTimes(2);
  },
  args: {
    color: 'primary',
  },
};

export const SecondaryButton: ComponentStoryObj<typeof Button> = {
  ...PrimaryButton,
  args: {
    color: 'secondary',
  },
};

export const WarningButton: ComponentStoryObj<typeof Button> = {
  ...PrimaryButton,
  args: {
    color: 'secondary',
  },
};

export const DangerButton: ComponentStoryObj<typeof Button> = {
  ...PrimaryButton,
  args: {
    color: 'danger',
  },
};
