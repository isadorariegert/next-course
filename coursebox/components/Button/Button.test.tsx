import React from 'react';
import { render, screen } from '../../test-utils';

import userEvent from '@testing-library/user-event';

import { Button } from './Button';
// in the test-library we have the render method that renders something on the screen;

describe('Button test cases', () => {
  it('Render check', () => {
    //in jest we can create a mock function
    const onClick = jest.fn();
    //for the render function we can use the "asFragment" function to detec our
    //render behavior providing a sintaxe the we gonna use to match snapshots
    const { asFragment } = render(<Button onClick={onClick}>Button</Button>);

    //comparing next and previous snapshot
    expect(asFragment()).toMatchSnapshot();
  });

  // it('Check onClick callback', () => {
  //   const onClick = jest.fn();

  //   render(<Button onClick={onClick}>button</Button>);

  //   //we can use the screen method to access DOM and find some element / make some actions ex:
  //   // screen.debug();
  //   //another ex:
  //   const element = screen.getByRole('button');

  //   //with the user event library we can emulate user actions
  //   userEvent.click(element);

  //   //here we check if our mockfunction is been called onClick
  //   expect(onClick).toHaveBeenCalled();
  // });
});
