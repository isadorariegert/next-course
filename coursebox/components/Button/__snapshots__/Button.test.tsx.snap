// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test cases Render check 1`] = `
<DocumentFragment>
  <button
    class="css-1htqu2z"
    color="primary"
  >
    Button
  </button>
</DocumentFragment>
`;
